src/estimation/src/BerdyHelper.cpp

```diff
@@ -1244,16 +1244,49 @@ bool BerdyHelper::computeBerdySensorMatrices(SparseMatrix<iDynTree::ColumnMajor>
     ////////////////////////////////////////////////////////////////////////
     if( m_options.includeAllJointTorquesAsSensors )
     {
-        for(DOFIndex idx = 0; idx < static_cast<DOFIndex>(m_model.getNrOfDOFs()); idx++)
+        if (m_options.berdyVariant == ORIGINAL_BERDY_FIXED_BASE)
         {
-            // Y for the joint torques is just a rows of 0s and one 1  corresponding to the location
+          for(DOFIndex idx = 0; idx < static_cast<DOFIndex>(m_model.getNrOfDOFs()); idx++)
+          {
+              // Y for the joint torques is just a rows of 0s and one 1  corresponding to the location
             // of the relative joint torques in the dynamic variables vector
             IndexRange sensorRange = this->getRangeDOFSensorVariable(DOF_TORQUE_SENSOR,idx);
             IndexRange jointTrqRange = this->getRangeDOFVariable(DOF_TORQUE,idx);
 
             matrixYElements.pushTriplet(Triplet(sensorRange.offset, jointTrqRange.offset, 1));
 
             // bY for the joint torques is zero
+          }
+        }
+        else 
+        {
+          assert(false);
```

Comment:
I guess you are still not compiling in debug. : )

Reply:
I am changing this to ```assert(m_options.berdyVariant == BERDY_FLOATING_BASE);```